### PR TITLE
Introduce AGL based demo product

### PR DESCRIPTION
**Pull request by iartemenko**

### CODE REVIEW

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux-libc-headers/linux-libc-headers_%.bbappend [hunk 1]
```diff
@@ -3,3 +3,5 @@ RENESAS_BSP_URL = "git://github.com/xen-troops/linux.git"
 BRANCH = "master"
 SRCREV = "${AUTOREV}"
 LINUX_VERSION = "4.14.75"
+
+SRC_URI = "${RENESAS_BSP_URL};protocol=git;nocheckout=0;branch=${BRANCH}"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux-libc-headers/linux-libc-headers_%.bbappend]: We do need explanation on why no checkout is needed: "nocheckout": Tells the fetcher to not checkout source code when unpacking when set to "1". Set this option for the URL where there is a custom routine to checkout code. The default is "0". IMO, this change is not obvious at all

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend [hunk 2]
```diff
@@ -4,6 +4,8 @@ require inc/xt_shared_env.inc
 
 RENESAS_BSP_URL = "git://github.com/xen-troops/linux.git"
 
+SRC_URI = "${RENESAS_BSP_URL};protocol=git;nocheckout=0;branch=${BRANCH}"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend]: ditto

File: recipes-domd/domd-image-weston/domd-image-weston.bbappend [hunk 3]
```diff
@@ -1,151 +0,0 @@
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/domd-image-weston.bbappend:1]: This is a comment to the whole patch: why do we need to remove weston images and the rest? What is the problem with them?

File: recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch [hunk 4]
```diff
@@ -0,0 +1,29 @@
+From 13abdd7af02592cf90f79b88b4ba3ebfa8fd9ee4 Mon Sep 17 00:00:00 2001
```
Comment (andr2000) [inline recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch:1]: With this patch you add so many unrelated recipes, so at least you should clearly describe what is the origin of those, why you add all at once etc. Please provide the rationale for such a big change

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend [hunk 5]
```diff
@@ -0,0 +1,3 @@
+SRC_URI_remove = " \
+    file://0001-lib-randutils.c-Fix-args-of-getrandom.patch \
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend:2]: Why it is not needed now? Will you also remove the patch itself?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend [hunk 6]
```diff
@@ -0,0 +1,3 @@
+SRC_URI_remove = " \
+    file://0001-lib-randutils.c-Fix-args-of-getrandom.patch \
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend:2]: ditto

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend [hunk 7]
```diff
@@ -18,6 +18,10 @@ SRC_URI_append_rcar = " \
     file://xen-chosen.dtsi;subdir=git/arch/${ARCH}/boot/dts/renesas \
 "
 
+SRC_URI_remove = " \
+    file://0004-ADSP-enable-and-add-sound-hardware-abstraction.patch \
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend:22]: Why it is not needed now? Will you also remove the patch itself?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-platform/images/agl-cluster-demo-platform.bbappend [hunk 8]
```diff
@@ -1,3 +1,3 @@
 require inc/agl-image.inc
 
-inherit populate_sdk_qt5
+#inherit populate_sdk_qt5
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-platform/images/agl-cluster-demo-platform.bbappend]: please remove this useless comment

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-platform/images/agl-image-minimal.bbappend [hunk 9]
```diff
@@ -1,6 +0,0 @@
-require inc/agl-image.inc
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-platform/images/agl-image-minimal.bbappend]: is this change related to this patch?

File: recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc [hunk 10]
```diff
@@ -4,6 +4,12 @@ DISTRO_FEATURES_append = " pam"
 # For virtualization
 DISTRO_FEATURES_append = " virtualization"
 
+# For v4l2-renderer
+DISTRO_FEATURES_append = " v4l2-renderer"
+
+# For pulseaudio
+DISTRO_FEATURES_append = " pulseaudio"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc:11]: You already added pulse audio [here](https://github.com/xen-troops/meta-xt-prod-tu2019-demo/pull/1/commits/0dc1b7f63e8a7b91bd56a9a8a78dae7355d7cbb0#diff-1f9a10db038f572c8431354c623926dbR17)
Any reason we have pulseaudio added twice?

File: recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc [hunk 11]
```diff
@@ -31,12 +31,12 @@ BBMASK = "meta-renesas/meta-rcar-gen3/recipes-multimedia/gstreamer"
 # Add for gstreamer plugins ugly
 LICENSE_FLAGS_WHITELIST = "commercial"
 
-PREFERRED_VERSION_gstreamer1.0 = "1.12.2"
-PREFERRED_VERSION_gstreamer1.0-libav = "1.12.2"
-PREFERRED_VERSION_gstreamer1.0-plugins-base = "1.12.2"
-PREFERRED_VERSION_gstreamer1.0-plugins-bad = "1.12.2"
-PREFERRED_VERSION_gstreamer1.0-plugins-good = "1.12.2"
-PREFERRED_VERSION_gstreamer1.0-plugins-ugly = "1.12.2"
+PREFERRED_VERSION_gstreamer1.0 = "1.14.4"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc]: And the reason for that is? If this is because of the fact that new AGL release uses newer version of gstreamer, then why do we set it here at all? Can we simply remove all the above and rely on what AGL wants?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/displaymanager/files/display_manager.conf [hunk 12]
```diff
@@ -3,3 +3,4 @@
                 <allow send_destination="com.epam.DisplayManager"/>
         </policy>
 </busconfig>
+
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/displaymanager/files/display_manager.conf]: unrelated change - please remove

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/display-manager.service [hunk 13]
```diff
@@ -1,14 +1,12 @@
...
-Requires=dbus.service
-Before=HomeScreen.service
+Requires=dbus.service weston-ready.service
+After=weston-ready.service
 
 [Service]
+Type=simple
 Environment="XDG_RUNTIME_DIR=/run/user/0"
-Type=dbus
-BusName=com.epam.DisplayManager
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/display-manager.service]: why did you delete this? is it not needed anymore?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-h3.cfg [hunk 14]
```diff
@@ -30,18 +38,19 @@ layers = (
...
-        height = 576;
-        destination = {
-            width = 1024;
-            height = 576;
-            x = 28;
-            y = 28;
+        layer = "navigation";
+        width = 800;
+        height = 480;
+	destination = {
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-h3.cfg]: tab?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-m3.cfg [hunk 15]
```diff
@@ -29,21 +36,21 @@ layers = (
...
-        height = 576;
-        destination = {
-            width = 1024;
-            height = 576;
-            x = 28;
-            y = 28;
+        layer = "navigation";
+        width = 800;
+        height = 480;
+	destination = {
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-m3.cfg]: tab?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb [hunk 16]
```diff
@@ -0,0 +1,12 @@
+DESCRIPTION = "C++ header only log"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb:1]: this recipe is already used by this [patch](https://github.com/xen-troops/meta-xt-prod-tu2019-demo/pull/1/commits/b4aca359919a3e18b91c5f4fc4b69b2d5b50719c) which comes before xt-log. please sort out the order of patches

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend [hunk 17]
```diff
@@ -0,0 +1,74 @@
+FILESEXTRAPATHS_append := ":${THISDIR}/weston"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend]: please describe the reason why such a split is needed

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston_%.bbappend [hunk 18]
```diff
@@ -13,77 +13,8 @@ python __anonymous () {
 }
 
 SRC_URI_append = "file://weston-seats.rules \
-                  file://add_screen_remove_layer_API.patch \
-                  file://0001-v4l2-renderer-Release-dma-buf-when-attaching-null-bu.patch \
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston_%.bbappend]: any reason these patches is removed? how is this change related to files being split?
If you remove these from SRC_URI then there is little reason to keep the files itself?

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend [hunk 19]
```diff
@@ -22,7 +22,7 @@ python __anonymous () {
 ################################################################################
 # Generic ARMv8
 ################################################################################
-SRC_URI = "repo://github.com/xen-troops/manifests;protocol=https;branch=master;manifest=prod_devel/dom0.xml;scmdata=keep"
+SRC_URI = "repo://github.com/xen-troops/manifests;protocol=https;branch=master;manifest=prod_tu2019_demo/dom0.xml;scmdata=keep"
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend:25]: these two changes are not related. commit message is only relevant for the second one

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domd-install-artifacts/domd-install-artifacts.bb [hunk 20]
```diff
@@ -15,7 +15,7 @@ do_install() {
    find "${XT_DIR_ABS_SHARED_BOOT_DOMD}" -iname 'Image*' -exec \
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domd-install-artifacts/domd-install-artifacts.bb:15]: please fix commit message:
s/varibale/variable
"so meet the changes" sounds strange ;)

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend [hunk 21]
```diff
@@ -141,17 +141,9 @@ do_deploy_append() {
     for DTB in ${KERNEL_DEVICETREE}
         do
               DTB_BASE_NAME=`basename ${DTB} | awk -F "." '{print $1}'`
-              DTB_NAME=`echo ${KERNEL_IMAGE_SYMLINK_NAME} | sed "s/${MACHINE}/${DTB_BASE_NAME}/g"`
+              DTB_NAME=`echo ${KERNEL_IMAGE_LINK_NAME} | sed "s/${MACHINE}/${DTB_BASE_NAME}/g"`
               DTB_SYMLINK_NAME=`echo ${DTB_NAME##*-}`
-              if [ -z "${KERNEL_IMAGETYPES}" ] ; then
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend:140]: why this part was deleted? how is it related to KERNEL_IMAGE_LINK_NAME change?

File: recipes-domd/agl/files/meta-xt-prod-extra/inc/agl-image.inc [hunk 22]
```diff
@@ -2,6 +2,7 @@ IMAGE_INSTALL_append = " \
     guest-addons \
     guest-addons-bridge-up-notification-service \
     guest-addons-bridge-config \
+    guest-addons-xendriverdomain-service \
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/inc/agl-image.inc:5]: "update guest addons" patches contain so many unrelated changes that I would love them to be split...

**Review by andr2000 — COMMENTED** [review]
the patches look great, but still need some improvement

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend [hunk 23]
```diff
@@ -1,74 +0,0 @@
-FILESEXTRAPATHS_append := ":${THISDIR}/weston"
```
Comment (andr2000) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend]: was this bbappend added when you split weston recipe and now you remove it?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend (hunk 2 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend]: Done

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux-libc-headers/linux-libc-headers_%.bbappend (hunk 1 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux-libc-headers/linux-libc-headers_%.bbappend]: Done

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend (hunk 5 above)
Comment (iartemenko) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend:2]: The patch already applied in current version.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston_%.bbappend (hunk 18 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston_%.bbappend]: The same with those patches.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend (hunk 5 above)
Comment (iartemenko) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/util-linux/util-linux_%.bbappend:2]: But it is still needed for other products which use poky less then 'thud'.
The patch itself is in the meta-xt-images layer and will not be removed for now.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/displaymanager/files/display_manager.conf (hunk 12 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/displaymanager/files/display_manager.conf]: Done

File: recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc (hunk 10 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-agl-bsp/conf/include/agl_rcar_xt.inc:11]: Deleted from "there". Leaving here in a single place.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend (hunk 7 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend:22]: It is related to linux kernel which is used by AGL.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-h3.cfg (hunk 14 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-h3.cfg]: fixed

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-m3.cfg (hunk 15 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/dm-salvator-x-m3.cfg]: fixed

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend (hunk 17 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend]: weston config generation was changed, even more,
this file going to be deleted in a several commits later

File: recipes-domd/agl/files/meta-xt-prod-extra/inc/agl-image.inc (hunk 22 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/inc/agl-image.inc:5]: Done

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend (hunk 23 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend]: as i mentioned above weston config generation was changed, so finally this recipe became unneeded.

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend (hunk 19 above)
Comment (iartemenko) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend:25]: splitted out

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domd-install-artifacts/domd-install-artifacts.bb (hunk 20 above)
Comment (iartemenko) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/domd-install-artifacts/domd-install-artifacts.bb:15]: Done

**Comment by iartemenko** [comment]
Done. [WIP] label removed.

File: recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch (hunk 4 above)
Comment (lorc) [reply recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch:1]: I second that.

For now this looks like a huge copy-paste across projects. They will diverge in the future and we will have hard time trying to backport fixes to and fro.

Maybe it is worth to include those files as a separate layer from the original repository?

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend (hunk 23 above)
Comment (iartemenko) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-graphics/wayland/weston-conf.bbappend]: squashed

File: recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch (hunk 4 above)
Comment (andr2000) [reply recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch:1]: Those come from different meta layers AFAIK, so it might not be a good idea to include. What is more is that we do not have AGL based development product for daily use, we only integrate AGL on purpose. This is why, for example, CES demos have AGL integrated and prod-devel not. So, I am not convinced we should "include" AGL, but instead directly use it as it is done in this series.

Comment (lorc) [reply recipes-domd/agl/files/0001-Do-not-try-to-use-XDG-environment-while-setting-up.patch:1]: Yes, but how to cope with fixes that were done for one AGL-based product, but weren't propaged to other ones?
